app/code/Magento/PageCache/view/frontend/templates/inline_form_key_provider.phtml

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+use Magento\Framework\View\Helper\SecureHtmlRenderer;
+
+$scriptString = <<<HTML
+    function setFormKeyCookie(value) {
+        var expires,
+            secure,
+            date = new Date(),
+            isSecure = !!window.cookiesConfig && window.cookiesConfig.secure;
+        date.setTime(date.getTime() + 86400000);
+        expires = '; expires=' + date.toUTCString();
+        secure = isSecure ? '; secure' : '';
+
+        document.cookie = 'form_key=' + (value || '') + expires + secure + '; path=/';
+    }
+    function getFormKeyCookie() {
+        var nameEQ = 'form_key=',
+            cookieArr = document.cookie.split(';');
+        for (var i = 0; i < cookieArr.length; i++) {
+            var cookie = cookieArr[i];
+            while(cookie.charAt(0) == ' ') {
+                cookie = cookie.substring(1, cookie.length);
+            };
+            if (cookie.indexOf(nameEQ) == 0) {
+                return cookie.substring(nameEQ.length, cookie.length);
+            };
+        }
+        return null;
+    }
+    function generateCookieString() {
+        var result = '',
+            length = 16;
+            chars = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+        while (length--) {
+           result += chars[Math.round(Math.random() * (chars.length - 1))];
+        }
+        return result;
+    }
+    var formKey = getFormKeyCookie();
+    if (!formKey) {
+        formKey = generateCookieString();
+        setFormKeyCookie(formKey);
+    }
```

Comment:
Could you not expose all these functions to the global scope, but only the needed (window.formKey)?